languageserver/src/main/scala/langserver/core/Connection.scala

```diff
@@ -153,12 +171,21 @@ abstract class Connection(inStream: InputStream, outStream: OutputStream)(implic
   }
 
   private def handleCommand(method: String, id: CorrelationId, command: ServerCommand): Future[Unit] = {
-    commandHandler(method, command).map { result =>
+    val future = commandHandler(method, command).map { result =>
       val rJson = ResultResponse.write(result, id)
       msgWriter.write(rJson)
     }.onErrorRecover {
       case NonFatal(e) =>
         logger.error(e.getMessage, e)
     }.runAsync
+    id match {
+      case NumericCorrelationId(value) =>
```

Comment:
I'll report the comment as a review comment, so it's easier to see.
Under what circumstances is this a `StringCorrelationId`? Should we ignore it?

Reply:
I've seen https://github.com/scalameta/language-server/pull/128#commitcomment-26300800, that makes sense.